src/test/librbd/test_librbd.cc

```diff
@@ -2211,15 +2227,36 @@ TEST_F(TestLibRBD, FlushAio)
   for (i = 0; i < num_aios; ++i) {
     ASSERT_EQ(0, rbd_aio_create_completion(NULL, NULL, &write_comps[i]));
     uint64_t offset = rand() % (size - TEST_IO_SIZE);
-    ASSERT_EQ(0, rbd_aio_write(image, offset, TEST_IO_SIZE, test_data,
-			       write_comps[i]));
+    ASSERT_EQ(0, rbd_aio_write(image, offset, TEST_IO_SIZE, test_data, write_comps[i]));
   }
 
   rbd_completion_t flush_comp;
   ASSERT_EQ(0, rbd_aio_create_completion(NULL, NULL, &flush_comp));
   ASSERT_EQ(0, rbd_aio_flush(image, flush_comp));
   ASSERT_EQ(0, rbd_aio_wait_for_complete(flush_comp));
   ASSERT_EQ(1, rbd_aio_is_complete(flush_comp));
+
+  // test with journaling enabled
```

Comment:
You don't need to add subsections within a test for various features.  We run "unittest_librbd" with numerous feature combinations with each rebuild under 'make check':

https://github.com/ceph/ceph/blob/master/src/test/run-rbd-unit-tests.sh